### ISSUE TITLE
soterdash display all the blocks in dotted line.

### ISSUE THREAD

**Issue by totaloutput** [soteria-dag/soterd#8]
the blocks in soterdash is now dotted line, it was solid line before the commit: https://github.com/soteria-dag/soterdash/commit/3047fcd90231428f212a8a484eb92356d4e9e5b6

here is a screenshot of it:
![image](https://user-images.githubusercontent.com/43481958/57942311-f9e48f80-7885-11e9-9ba9-302cfaf7983e.png)

**Comment by colakong** [comment]
This was a [regression](https://github.com/soteria-dag/soterd/commit/fc811bc59404c51b691e22d5b670f8299999457b#diff-2ddd58f7613c0e0b58f9c7cadc1dd2dcR159) introduced in fc811bc59404c51b691e22d5b670f8299999457b.

* dag block order calculation is bottom-up (from genesis if there's no caching)
* dag block colouring calculation is top-down (from tips)

So by re-using the result of an old call to `calculateBlueSet`, we're not considering the latest 200 blocks in the colouring, which makes it more likely to get colouring wrong.

I wrote a test to identify differences in colouring between `OrderDAG` with and without caching. The test would most often fail, because a [new virtual node](https://github.com/soteria-dag/soterd/blob/fc811bc59404c51b691e22d5b670f8299999457b/blockdag/phantom/coloring.go#L146) would be [added to blueSetCache](https://github.com/soteria-dag/soterd/blob/fc811bc59404c51b691e22d5b670f8299999457b/blockdag/phantom/coloring.go#L102) on each call to `OrderDAG` (every generation). As a result, `blueSetCache` would have multiple _VIRTUAL_ node entries, one for each generation, so when comparing `blueSetCache` between `OrderDAG` and `OrderDAGNoCache`, it would often compare _VIRTUAL_ node entries from different generations, and fail because they don't have the same contents.

I've prepared a patch to fix the two issues. @jenlouie8's reviewing it, so we can decide what to do about the _VIRTUAL_ node entries (in my patch I had it so that they aren't included in `blueSetCache`). If it looks good we'll merge and push, and this bug will be fixed :).
